README.md

```diff
@@ -31,6 +34,16 @@ Modern software development practices _assume_ support for reviewing changes, tr
 - [Gitkube](https://gitkube.sh/) - Build and deploy docker images to Kubernetes using git push
 - [Weave Cloud](https://www.weave.works/product/cloud/) - GitOps experience, workflows and dashboard for your cluster(s) (commercial product from Weaveworks)
 
+## Public Cloud Providers
+
+### Google Cloud
+
+- [Cloud Build](https://cloud.google.com/kubernetes-engine/docs/tutorials/gitops-cloud-build) GitOps-style continuous delivery with Cloud Build
```

Comment:
This looks to me like the opposite of GitOps - more like [CiOps](https://www.weave.works/blog/kubernetes-anti-patterns-let-s-do-gitops-not-ciops)

Reply:
Well it's not black and white which is which.  I've done exactly what this article describes before, and from a user's perspective it's exactly the same.  I commit to my repo (in my case it was on cloud source repositories) and it gets deployed to kubernetes almost immediately after.

Maybe this conversation is demonstrating a need to define what specific characteristics are necessary to qualify?

Reply:
I think Ilya explains it really well in the article I've linked above. Running `kubectl apply` from outside the cluster .e.g some CI doesn't make it GitOps in my opinion. If we go down this road let's add all the CI SaaS offerings to this repo, since all of them can shell out to kubectl.

Reply:
I've read Ilya's article (although, perhaps I should read it again since the first go was some time ago now), so forgive me for asking... but [shelling out to kubectl is "all" flux does](https://github.com/weaveworks/flux/blob/master/cmd/fluxd/main.go#L410), I'm not clear on the consequences of the distinction you're trying to make.

To say it another way, I understand that you're saying the boundary needs to not extend past the point of the cluster... but I can perfectly well run these tools within my cluster.  For example the GitLab CI can be run in cluster (_as could Cloud Build_ if you and I both worked the GCP)..  That said, I feel like we would agree that doesn't qualify it as a GitOps tool.

This is probably the wrong place for us to have this discussion... but all the same..  I bet I'm not going to be the last person to not be clear about this.

Reply:
You are right, I'm biased towards controllers, the idea that the desired state is continuously  reconciled. A CI system is reactive by design, it gets triggered by a Git hook, while a controller would detect changes in the cluster and reapply the state to correct the configuration drift.

Reply:
> A CI system is reactive by design, it gets triggered by a Git hook
I don't see a big difference here between a git hook and a container lifecycle hook.  As far as I'm concerned Flux is reactive as well...  In fact, you might even say that it's worse than reactive because it only works in a 30 second loop (by default) scanning the repo and scanning kubernetes (as compared to a system where it is completely event driven - where the events would be git hooks and kubernetes hooks).

Not to mention that the problem you're hitting on isn't _inherent_ to a CI.  That is to say.. you could make a CI/CD which doesn't suffer from problems with drift (as a design goal).  I think [Spinnaker](https://www.spinnaker.io/) is a living example of this.

anyway.  I don't want to get too deep into the weeds here.  If you think these don't belong please just let me know and I'll happily close the PR.

I value and trust your opinion.

Reply:
I think the Cloud Build article is better placed in the tutorials section, as for the AWS project I don't think it qualifies.

Reply:
thanks.  on it.